src/components/Calculator.js

```diff
@@ -1,46 +1,56 @@
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+    };
+    this.handler = this.handler.bind(this);
+  }
+
+  handler(e) {
+    this.setState((state) => calculate(state, e.target.innerText));
   }
 
   render() {
+    const { next, total } = this.state;
     return (
       <table className="TableContainer">
         <tr>
```

Comment:
* Kindly fix the warning: validate Dom nesting, I get this on page load fixing this would make your project look professional
![image](https://user-images.githubusercontent.com/52497246/130104856-582f2861-5efe-42f1-b003-60ef5a93c1d0.png)